bundles/org.openhab.binding.pulseaudio/src/main/java/org/openhab/binding/pulseaudio/internal/PulseAudioAudioSource.java

```diff
@@ -0,0 +1,250 @@
+/**
+ * Copyright (c) 2010-2022 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.pulseaudio.internal;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.PipedInputStream;
+import java.io.PipedOutputStream;
+import java.net.Socket;
+import java.util.HashSet;
+import java.util.Set;
+import java.util.concurrent.Future;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.function.Consumer;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.openhab.binding.pulseaudio.internal.handler.PulseaudioHandler;
+import org.openhab.core.audio.AudioException;
+import org.openhab.core.audio.AudioFormat;
+import org.openhab.core.audio.AudioSource;
+import org.openhab.core.audio.AudioStream;
+import org.openhab.core.common.ThreadPoolManager;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The audio source for openhab, implemented by a connection to a pulseaudio source using Simple TCP protocol
+ *
+ * @author Miguel Álvarez - Initial contribution
+ *
+ */
+@NonNullByDefault
+public class PulseAudioAudioSource extends PulseaudioSimpleProtocolStream implements AudioSource {
+
+    private final Logger logger = LoggerFactory.getLogger(PulseAudioAudioSource.class);
+    private final Set<PipedOutputStream> pipeOutputs = new HashSet<>();
+    private final ScheduledExecutorService executor = ThreadPoolManager
+            .getScheduledPool("OH-binding-pulseaudio-source");
```

Comment:
The convention is to use the thing UID in the name. If I correctly remember OH-<thingUID>-xxx
You have the thing handler, so you can the thing UID.

Reply:
I am sorry, after checking, the naming convention is `OH-binding-<thingUID>-xxx` with the last part being optional.

And I think `executor` should be set in the constructor when `pulseaudioHandler` is defined.